Update Product test_positive_delete_1 CLI test

elyezer opened the pull request:
The test is failing in automation because the info command are being
issued immediately the delete command, but after some time the product
is really deleted and the info returns the expected result.

In order to avoid this failure, add workaround to sleep for a few
seconds before running the info command but just if BZ #1219490 is open.

elyezer commented:
Rebased

oshtaier commented:
Personally, hate to use any static delays in test automation and trying to find another solution *literally always*. Do we have guarantee that 5 seconds will be enough on all systems under test?
Will appreciate if you confirm that it was only one possible way to resolve that issue

Ichimonji10 commented:
Unfortunately, I have seen solutions like this in several places. The solutions like this are used because the server lies about its own state. A slightly more robust solution may be to poll the server several times after deleting a product:

```python
if bz_bug_is_open(1219490):
    for _ in range(5):
        if Product.info({…}).return_code == 0:
            sleep(5)
        else:
            break
```

Ichimonji10 commented:
ACK pending response.

oshtaier commented:
Can we make some kind of generic function that will do server 'ping' or something similar and which we will use in situation like this?
Just today we need sleep 5 * 5 and tomorrow we will need sleep 10 * 10. It will be good to move that things in one place. And in case we have too many usages of that function, even move that sleep value into our 'robottelo.properties' file

elyezer commented:
This kind of thing should not be be present on robottelo, this is just something to avoid bad test results. I can skip the entire test but I prefer the workaround in this case because it will ensure the functionality even by "accepting" the wrong state.

I will do the polling like suggested. Related the generic approach, we need to research more about this because API tests differs from CLI tests that both differs from UI.

oshtaier commented:
Sure. I am totally fine to go further with mentioned approach, but really appreciate if we can create separate issue for such investigation with at least 'Low' priority

sthirugn commented:
ACK. But I agree to @oshtaier we need to revisit this in future with more reliable solution.

elyezer commented:
@oshtaier feel free to open the issue. I agree on investigating more, this could result in a generic approach that will help us doing these workarounds in the future.

elyezer commented:
Updated and rebased.

```
py.test tests/foreman/cli/test_product.py -ktest_positive_delete_1
=================================== test session starts ===================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 53 items

tests/foreman/cli/test_product.py .

==================== 52 tests deselected by '-ktest_positive_delete_1' ====================
======================== 1 passed, 52 deselected in 24.27 seconds =========================
```

oshtaier commented:
@elyezer Created issue #2198 per our agreement

Ichimonji10 commented:
The code itself has been reworked to make use of polling, an issue has been created asking for a more robust solution to this problem and we have two ACKs. Merging.
